packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -730,6 +764,51 @@ class _TextSelectionGestureDetectorState extends State<TextSelectionGestureDetec
     }
   }
 
+  DragStartDetails _lastDragStartDetails;
+
+  void _handleDragStart(DragStartDetails details) {
+    assert(_lastDragStartDetails == null);
+    _lastDragStartDetails = details;
+    if (widget.onDragSelectionStart != null) {
+      widget.onDragSelectionStart(details);
+    }
+  }
+
+  Timer _dragUpdateTimer;
+  DragUpdateDetails _lastDragUpdateDetails;
+
+  void _handleDragUpdate(DragUpdateDetails details) {
+    _lastDragUpdateDetails = details;
+    // Only schedule a new timer if there's no one pending.
+    _dragUpdateTimer ??= Timer(_kDragSelectionUpdateThrottle, _handleDragUpdateThrottled);
```

Comment:
Pass it a closure? Seems cleaner than maintaining state around instance variables.

Reply:
I wanted to use a closure here too, but it's not possible because `_handleDragUpdateThrottled` is also called from `_handleDragEnd`.